utils/script-luajit.c

```diff
@@ -0,0 +1,425 @@
+#include <lauxlib.h>
+#include "utils/utils.h"
+#include "utils/symbol.h"
+#include "utils/fstack.h"
+#include "utils/filter.h"
+#include "utils/script.h"
+#include "utils/script-luajit.h"
+
+static const char *libluajit = "libluajit-5.1.so";
+static void *luajit_handle;
+static lua_State *L;
+
+static lua_State* (*dlluaL_newstate)(void);
+static void (*dlluaL_openlibs)(lua_State *L);
+static int (*dlluaL_loadfile)(lua_State *L, const char *filename);
+static void (*dllua_close)(lua_State *L);
+static int (*dllua_pcall)(lua_State *L, int nargs, int nresults, int errfunc);
+static int (*dllua_next)(lua_State *L, int index);
+static void (*dllua_createtable)(lua_State *L, int narr, int nrec);
+static void (*dllua_gettable)(lua_State *L, int index);
+static void (*dllua_settable)(lua_State *L, int index);
+static const char* (*dllua_tolstring)(lua_State *L, int index, size_t *len);
+static void (*dllua_pushstring)(lua_State *L, const char *s);
+static void (*dllua_pushinteger)(lua_State *L, lua_Integer n);
+static void (*dllua_pushnumber)(lua_State *L, lua_Number n);
+static void (*dllua_pushboolean)(lua_State *L, int b);
+static void (*dllua_pushnil)(lua_State *L);
+static void (*dllua_remove)(lua_State *L, int index);
+
+static void (*dllua_getfield)(lua_State *L, int index, const char *k);
+static int (*dllua_type)(lua_State *L, int index);
+static void (*dllua_settop)(lua_State *L, int index);
+
+void __attribute__((weak)) lua_getfield(lua_State *L, int index, const char *k)
+{
+	dllua_getfield(L, index, k);
+}
+
+int __attribute__((weak)) lua_type(lua_State *L, int index)
+{
+	return dllua_type(L, index);
+}
+
+void __attribute__((weak)) lua_createtable(lua_State *L, int narr, int nrec)
+{
+	dllua_createtable(L, narr, nrec);
+}
+
+const __attribute__((weak)) char *lua_tolstring(lua_State *L, int index, size_t *len)
+{
+	return dllua_tolstring(L, index, len);
+}
+
+void __attribute__((weak)) lua_settop(lua_State *L, int index)
+{
+	return dllua_settop(L, index);
+}
+
+static void setup_common_context(struct script_context *sc_ctx)
+{
+	lua_newtable(L); // * ctx
+	dllua_pushstring(L, "tid"); // * ctx 'tid'
+	dllua_pushinteger(L, sc_ctx->tid); // * ctx 'tid' tid
+	dllua_settable(L, -3); // * ctx
+	dllua_pushstring(L, "depth"); // * ctx 'depth'
+	dllua_pushinteger(L, sc_ctx->depth); // * ctx 'depth' depth
+	dllua_settable(L, -3); // * ctx
+	dllua_pushstring(L, "timestamp"); // * ctx 'timestamp'
+	dllua_pushinteger(L, sc_ctx->timestamp); // * ctx 'timestamp' timestamp
+	dllua_settable(L, -3); // * ctx
+	dllua_pushstring(L, "duration"); // * ctx 'duration'
+	dllua_pushinteger(L, sc_ctx->duration); // * ctx 'duration' duration
+	dllua_settable(L, -3); // * ctx
+	dllua_pushstring(L, "address"); // * ctx 'address'
+	dllua_pushinteger(L, sc_ctx->address); // * ctx 'address' address
+	dllua_settable(L, -3); // * ctx
+	dllua_pushstring(L, "name"); // * ctx 'name'
+	dllua_pushstring(L, sc_ctx->name); // * ctx 'name' name
+	dllua_settable(L, -3); // * ctx
```

Comment:
All these comments look not so meaningful.  Also please use C-style comments if possible.